prereqs.py

```diff
@@ -157,60 +156,134 @@ def check_prereqs(install=False, skip_wasm=False):
-        wasm_minor = int(version_match.group(2))
-        if wasm_major != wasmpack_ver[0] or wasm_minor < wasmpack_ver[1]:
+        found_ver = tuple(int(g) for g in version_match.groups())
+        if found_ver < wasm_bindgen_ver:
             print(
-                f"wasm-pack version must be {wasmpack_ver[0]}.{wasmpack_ver[1]} or later. Please update."
+                f"wasm-bindgen v{wasm_bindgen_ver[0]}.{wasm_bindgen_ver[1]}.{wasm_bindgen_ver[2]} or later is required. Please update."
             )
             exit(1)
     else:
-        print("Unable to determine the wasm-pack version")
+        print("Unable to determine the wasm-bindgen version")
+
+    ### Check the binaryen version ###
+    try:
+        binaryen_version = subprocess.check_output(["wasm-opt", "--version"])
+        print(f"Detected wasm-opt version: {binaryen_version.decode()}")
+    except FileNotFoundError:
+        if install == True:
+            print("wasm-opt not found. Attempting to install...")
+            install_binaryen()
+            binaryen_version = subprocess.check_output(["wasm-opt", "--version"])
+        else:
+            print(
+                "wasm-opt not found. Install via 'python ./prereqs.py --install' or see https://github.com/WebAssembly/binaryen"
+            )
+            exit(1)
+    version_match = re.search(r"wasm-opt version (\d+)", binaryen_version.decode())
+    if version_match:
+        found_ver = int(version_match.group(1))
+        if found_ver < binaryen_ver:
+            print(f"wasm-opt version must be {binaryen_ver} or later. Please update.")
+            exit(1)
+    else:
+        print("Unable to determine the wasm-opt version")
+        exit(1)
 
     # Ensure the required wasm target is installed
     if "wasm32-unknown-unknown" not in installed_rust_targets:
-        print("WASM rust target is not installed.")
-        print("Please install the missing target by running:")
-        print("rustup target add wasm32-unknown-unknown")
+        if install == True:
+            print("Wasm Rust target not installed. Attempting to install...")
+            subprocess.run(
+                ["rustup", "target", "add", "wasm32-unknown-unknown"], check=True
+            )
+        else:
+            print(
+                "Wasm Rust target not installed. Install via 'rustup target add wasm32-unknown-unknown'"
+            )
+            exit(1)
+
+
+def download_and_extract(url_base, tar_file, out_dir):
+    os.makedirs(out_dir, exist_ok=True)
+    temp_file = tempfile.gettempdir() + os.sep + tar_file
+
+    # Note: Using curl and tar as subprocesses rather than Python libraries for features such as --strip-components
+    subprocess.run(["curl", "-L", "-o", temp_file, url_base + tar_file], check=True)
+    subprocess.run(
+        ["tar", "-xzf", temp_file, "--strip-components=1", "-C", out_dir], check=True
+    )
+
+    os.remove(temp_file)  # Clean up the tar file
+
+
+def install_wasm_bindgen():
```

Comment:
What about `cargo install wasm-bindgen-cli` with `--locked` and a specific version?

Reply:
`cargo binstall` isn't availalbe by default though, so that's another prereq instead.

> I trust building from source the most

Maybe we should recommend users get the QDK by building the repo themselves then instead of installing from the marketplace 😛

Reply:
In all seriousness though, the bininstall page states _"Binstall works by fetching the crate information from crates.io and searching the linked repository for matching releases and artifacts"_, so even if we did the work to use it, it's probably unpacking exactly the same tarball as an end result.